File: app/app/src/main/java/me/williamhester/reddit/inject/ApplicationComponent.java
package me.williamhester.reddit.inject;

import javax.inject.Singleton;

import dagger.Component;
import me.williamhester.reddit.ui.activities.BaseActivity;
import me.williamhester.reddit.ui.fragments.BaseFragment;

/**  */
@Component(modules = {ApiModule.class, ApplicationModule.class})
@Singleton
public interface ApplicationComponent {
  void inject(BaseFragment fragment);
  void inject(BaseActivity activity);
}

File: app/app/src/main/java/me/williamhester/reddit/inject/ApplicationComponent.kt
package me.williamhester.reddit.inject

import dagger.Component
import javax.inject.Singleton

/** The Dagger component for injecting the ApiModule and ApplicationModules. */
@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {
  /** Factory method for creating the ActivityComponent. */
  fun plus(activityModule: ActivityModule) : ActivityComponent
}
